modules/migrations/gogs_test.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2019 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package migrations
+
+import (
+	"testing"
+	"time"
+
+	"code.gitea.io/gitea/modules/migrations/base"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGogsDownloadRepo(t *testing.T) {
+	downloader := NewGogsDownloader("https://try.gogs.io", "c109b3c905eb57951cfdea270cfcfdc297a74500", "", "lunnytest", "TESTREPO")
+	repo, err := downloader.GetRepoInfo()
+	assert.NoError(t, err)
```

Comment:
What happens if try.gogs.io is down, or your repo gets remove...will all builds then fail? I know we already rely on github and other sites being available for building but maybe for these type of migration test it can skip them if the URL doesn't respond properly right away. Even now try.gogs.io is very slow for me and times out on some pages.

Reply:
Agree, Internet-dependent tests should ideally not run during regular CI, but only when explicitely triggered e.g. `make test-internet` or similar.